SPEC.md

```diff
@@ -2601,6 +2602,20 @@ Integer ylen = length(ys) # 3
 Integer zlen = length(zs) # 0
 ```
 
+## Array[X] flatten(Array[Array[X]])
+
+Given an array of arrays, the `flatten` function concatentes all the
+member arrays in the order to appearance to give the result. For example:
+
+```
+Array[Array[Integer]] ai2D = [[1, 2, 3], [11], [21,22]]
+Array[Integer] ai = flatten(ai2D)   # [1, 2, 3, 11, 21, 22]
+
+
+Array[Array[File]] af2D = [["/tmp/X.txt"], ["/tmp/Y.txt", "/tmp/Z.txt"], []]
+Array[Integer] af = flatten(af2D)   # ["/tmp/X.txt", "/tmp/Y.txt", "/tmp/Z.txt"]
```

Comment:
Does this also work with `Array`-likes? eg maps?
```wdl
Array[Map[Int, String]] array_of_maps = [ {1: "one"}, {2: "two", 3: "three"} ]
Array[Pair[Int, String]] array_of_pairs = flatten(array_of_maps) # [ (1, "one"), (2, "two"), (3, "three") ]
```

Reply:
Honestly, I did not consider what would happen if a user applied `flatten` to an array-like structure. It is a useful extension though, in my opinion. If we use the scala [flatten](http://www.scala-lang.org/api/2.12.3/scala/collection/immutable/List.html#flatten[B]:List[B]) semantics, we would get:
```
Array[Map[Int, String]] array_of_maps = [ {1: "one"}, {2: "two", 3: "three"} ]
Map[Int, String] m = flatten(array_of_maps) # {1: "one", 2: "two", 3: "three"}
```